main.go

```diff
@@ -133,6 +138,29 @@ func main() {
 	}
 
 	setupFinished := make(chan struct{})
+	go func() {
+		// Block until the setup (certificate generation) finishes.
+		setupLog.Info("waiting to create operator webhook for MetalLB CR")
+		<-setupFinished
+		setupLog.Info("creating operator webhook for MetalLB CR")
+		if err = (&metallbv1beta1.MetalLB{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "operator webhook", "MetalLB")
+			os.Exit(1)
+		}
+		setupLog.Info("operator webhook for MetalLB CR is created")
+
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.Handler())
```

Comment:
since we currently don't expose metrics, a readyz endpoint that just returns ok is enough.

Reply:
Also, now that we have a readyness check that says when the operator is available ot serve the webhooks, we can change the CI here https://github.com/metallb/metallb-operator/blob/ba046ddeca3d10413a40c970c5702cfd37bc62bb/.github/workflows/metallb_e2e.yml#L78 to wait for pods to be ready instead of having the sleep. This is going to make it more deterministic and less flaky